frontend/src/pages/Match/index.js

```diff
@@ -0,0 +1,99 @@
+import { useState } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import CancelOutlinedIcon from '@mui/icons-material/CancelOutlined';
+import PersonOutlineIcon from '@mui/icons-material/PersonOutline';
+import Button from '@mui/material/Button';
+import CircularProgress from '@mui/material/CircularProgress';
+import Grid from '@mui/material/Grid';
+import Typography from '@mui/material/Typography';
+
+import OutlinedContainer from '../../components/ui/OutlinedContainer';
+import styles from './Match.module.css';
+import Timer from './Timer';
+
+function MatchPage() {
+  const [isMatchFailed, setIsMatchFailed] = useState(false);
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const handleCancel = () => {
+    navigate('/home');
+  };
+  const handleRetry = () => {
+    setIsMatchFailed(false);
+  }
+
+  const darkGrey = '#707070';
```

Comment:
Possible optimization for rendering could be putting the const global variable outside of functional component, so they don't get re-rendered again when state change.

Reply:
Nice, never come across this optimization before. Thanks for the tip!